src/resources/addys/interactions.ts

```diff
@@ -0,0 +1,184 @@
+  fromAddress?: string;
+  toAddress: string;
+  currency?: NativeCurrencyKey;
+  chainId: number;
+};
+
+// ///////////////////////////////////////////////
+// Query Key
+
+// Query key remains the same, fetching all transactions for the given addresses
+export function interactionsCountQueryKey({
+  fromAddress,
+  toAddress,
+  currency,
+}: Required<Omit<InteractionsCountArgs, 'chainId'> & { currency: NativeCurrencyKey }>) {
+  return createQueryKey(
+    'interactionsCount',
+    { fromAddress: fromAddress.toLowerCase(), toAddress: toAddress.toLowerCase(), currency },
+    { persisterVersion: 1 }
+  );
+}
+
+type InteractionsCountQueryKey = ReturnType<typeof interactionsCountQueryKey>;
+
+const stableInteractionsCountResult: InteractionsCountResult = {};
+
+export async function interactionsCountQueryFunction({
+  queryKey,
+}: QueryFunctionContext<InteractionsCountQueryKey>): Promise<InteractionsCountResult> {
+  const [{ fromAddress, toAddress, currency }] = queryKey;
+  const supportedChainIds = useBackendNetworksStore.getState().getInteractionsWithSupportedChainIds();
+
+  if (!fromAddress || !toAddress || supportedChainIds.length === 0) {
+    logger.warn('[interactionsCountQueryFunction]: Missing address or supported chains, returning empty array.');
+    return stableInteractionsCountResult;
+  }
+
+  const url = `/${supportedChainIds.join(',')}/${fromAddress}/transactions`;
+
+  try {
+    const { data } = await getAddysHttpClient().get<InteractionsResponse>(url, {
+      params: {
+        currency: currency.toLowerCase(),
+        interactions_with: toAddress,
+      },
+      timeout: 20000,
+    });
+
+    if (data.meta.status !== 'ok' && data.meta.errors) {
+      logger.warn('[interactionsCountQueryFunction]: API returned errors for interactions count', {
+        apiErrors: data.meta.errors,
+        fromAddress,
+        toAddress,
+        chainIds: supportedChainIds,
+      });
+    }
+
+    return data.payload.transactions.reduce<InteractionsCountResult>((acc, tx) => {
+      acc[tx.chain_id] = acc[tx.chain_id] || 0;
+      acc[tx.chain_id]++;
+      return acc;
+    }, {});
+  } catch (error) {
+    logger.error(new RainbowError('[interactionsCountQueryFunction]: Failed to fetch interactions count'), {
+      message: (error as Error)?.message,
+      fromAddress,
+      toAddress,
+      currency,
+      chainIds: supportedChainIds,
+    });
+    return stableInteractionsCountResult;
+  }
+}
+
+export interface SelectedInteractionsCount {
+  totalCount: number;
+  specificChainCount: number;
+}
+
+// ///////////////////////////////////////////////
+// Query Hook
+
+export function useInteractionsCount(
+  { fromAddress: inputFromAddress, toAddress, currency: inputCurrency, chainId }: InteractionsCountArgs,
+  config: QueryConfigWithSelect<InteractionsCountResult, Error, SelectedInteractionsCount, InteractionsCountQueryKey> = {}
+) {
+  const { accountAddress, nativeCurrency } = useAccountSettings();
```

Comment:
nit: No strong opinion here, but should we consider deprecating use of `useAccountSettings` in scenarios where we can get that data from elsewhere since we are trying to move away from redux in the near future?

I think we can grab both of these values from `userAssetsStoreManager` now.

Reply:
Also, I'm not sure I understand the purpose of the `inputCurrency` param. In what scenario are we passing that in?

Reply:
I was told to not use that in standup yesterday and to rely on the useAccountSettings hook for the meantime.

Reply:
Let me ask again in standup, I might've heard incorrectly.